packages/auth/src/Authenticator.js

```diff
@@ -17,6 +19,39 @@ export class Authenticator {
     this.accessPolicyBuilder = new AccessPolicyBuilderClass(models);
   }
 
+  /**
+   * Authenticate by access token claims
+   * @param {{ userId: string, apiClientUserId?: string }} accessTokenClaims
+   */
+  async authenticateAccessToken({ userId, apiClientUserId }) {
+    const user = await this.models.user.findById(userId);
+    const userAccessPolicy = await this.getAccessPolicyForUser(user.id);
+
+    if (!apiClientUserId) {
+      return { user, accessPolicy: userAccessPolicy };
+    }
+
+    const apiClientAccessPolicy = await this.getAccessPolicyForUser(apiClientUserId);
+    return { user, accessPolicy: mergeAccessPolicies(userAccessPolicy, apiClientAccessPolicy) };
+  }
```

Comment:
I feel a touch unclear about what a the direct responsibilities of `Authenticator` vs the various authentication middlewares, etc. It seems that decrypting the `authHeader` is up to the middleware, but `Authenticator` handles how we associate users/accessPolicies with the headers... I may not have gotten the separation of concerns quite right here...

Reply:
Yeah this particular method seems a bit odd, because it doesn't actually check anything. All of the others verify that the authentication requirements are met, and then return the user's details and access policy, whereas this one assumes that the auth requirements have already been met externally. 

More consistent might be:
- Middleware pulls out just the token from the header (i.e. the `xxx` in `Bearer xxx`) 
- Passes that token in here, which calls `getTokenClaims` to validate the token and get the claims out, and then runs the existing content of this method